### PR TITLE
[DRAFT] Feature/mig 100 migrate migration program documentation v2

### PR DESCRIPTION
**Pull request by Ruslan05**

### CODE REVIEW

File: docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+Spryker comes pre-configured with the following services by default:
+
+* RabbitMQ
+* ElasticSearch
+* Redis
+* MariaDB
+* Jenkins
+
+After you have [upgraded the PHP version](/docs/scos/dev/migration-concepts/migrate-to-paas/step-2-upgrade-the-php-version.html), you can migrate these services without significant challenges, however, make sure to validate their versions. If your project uses components beyond the standard configuration, such as Kafka, we highly recommend to conduct thorough investigation.
```
Comment (lenadoc) [inline docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md:15]: "make sure to validate their versions" - are there any recommendations as to what versions should these services have?
"we highly recommend to conduct thorough investigation" - what specific investigation do you mean here?

File: docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Spryker comes pre-configured with the following services by default:
+
+* RabbitMQ
+* ElasticSearch
+* Redis
+* MariaDB
+* Jenkins
+
+After you have [upgraded the PHP version](/docs/scos/dev/migration-concepts/migrate-to-paas/step-2-upgrade-the-php-version.html), you can migrate these services without significant challenges. However, make sure to validate their versions. If your project uses components beyond the standard configuration, such as Kafka, we highly recommend conducting a thorough investigation.
```
Comment (lenadoc) [inline docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md:15]: "make sure to validate their versions" - are there any recommendations as to what versions should these services have?
"we highly recommend conducting thorough investigation" - what specific investigation do you mean here?

File: docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+For more information about the standard components and configuration, see [Spryker technology stack](/docs/scos/dev/architecture/technology-stack.html) and [conceptual overview](/docs/scos/dev/architecture/conceptual-overview.html).
+
+## Postgres to MariaDB
+
+Currently, within the Spryker PaaS environment, only the MariaDB engine is compatible with databases. Consequently, if the project previously relied on PostgreSQL, it is necessary to migrate to MariaDB. To accommodate this change, you need to make specific adjustments within the `config/Shared/config_*.php` files.
+
+Consider that CTE queries or any other raw SQL queries or DB structure specific to PostgreSQL have to be reviewed for compatibility with MariaDB. Also, it's a known fact that the CTE queries work slower on the MariDB-based engine.
```
Comment (lenadoc) [inline docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md:23]: in the "[step 13](docs/scos/dev/migration-concepts/migrate-to-paas/step-13-database-migration.md)" document below we already have dedicated information about migrating from Postgres to Maria DB. Does it make sense to you to remove this section completely from here then?

File: docs/scos/dev/migration-concepts/migrate-to-paas/step-13-migrate-database.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+```
+For additional information about this step, see [Importing data from a MariaDB or MySQL database to a MariaDB or MySQL DB instance](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MySQL.Procedural.Importing.SmallExisting.html).
+4. Upload the database dump to an AWS S3 bucket of the production environment using the AWS Management Console.
+5. Connect to the EC2 instance. For details on how to do this, see [Create an Amazon EC2 instance and copy the compressed database](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/MySQL.Procedural.Importing.NonRDSRepl.html#MySQL.Procedural.Importing.Import.Database).
+
+Optional: For the Spryker Cloud environment, we recommend using a Scheduler instance for backup/restore purposes.
+* AWS → EC2 → Instances → <env_name>-<[/d/w]*> → Connect via Session Manager
+  ![Scheduler Container example](https://spryker.s3.eu-central-1.amazonaws.com/docs/scos/dev/migration-concepts/migrate-to-paas/scheduler-container-example.png)
+* Find AWS credentials here:
+  ![AWS S3 credentials](https://spryker.s3.eu-central-1.amazonaws.com/docs/scos/dev/migration-concepts/migrate-to-paas/aws-s3-credentials.png)
```
Comment (lenadoc) [inline docs/scos/dev/migration-concepts/migrate-to-paas/step-13-migrate-database.md:48]: What do these steps mean? Can you elaborate more on each of them (what exactly is to be done and why)? As if you use just menu items, it is not very clear what exactly should be done and why

**Comment by lenadoc** [comment]
@Ruslan05 Are all of the steps (1-14) of this migration required? If some steps are optional and migration can happen without them, please list those steps

File: docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md (hunk 2 above)
Comment (Ruslan05) [reply docs/scos/dev/migration-concepts/migrate-to-paas/step-3-migrate-non-standard-services.md:15]: 1. Not really. So here we need to check if Spryker would work the same way if on-prem version of e.g. ElasticSearch is different from what Spryker Cloud provides. If the versions are too much different and app is not compatible with what Spryker Cloud offers then we should act toward solving this gap. So far i only found requirements for application, but not for Cloud: https://docs.spryker.com/docs/scos/dev/system-requirements/202311.0/system-requirements.html#base-shop-system-requirements
2. I mean that they should investigate how to solve this gap. For example they use Kafka, but Spryker doesn't offer it in Cloud. Therefore they either should think about replacement of Kafka with RabbitMQ or wire Kafka to Spryker Cloud somehow, e.g. using the VPN.

File: docs/scos/dev/migration-concepts/migrate-to-paas/step-13-migrate-database.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+7. To find the database credentials, filter the parameters by `DB`.      
+8. Restore snapshot to a new Spryker Cloud DB:
+
+```bash
+mysql -u<SPRYKER_DB_USERNAME> -p<SPRYKER_DB_PASSWORD> -h <Endpoint> <DB name> < <dump_name>.sql
+```
+9. Repeat step 2 against the restored database. To make sure the migration is successful, compare the outputs from the queries you've run for the original and for the restored DBs. They should be the same.
+
+## PostgreSQL to MariaDB
+PostgreSQL DB must be converted to MariaDB-compatible dump. More information can be found [here](https://mariadb.com/kb/en/migrating-to-mariadb-from-postgresql/).
```
[the remark this answers is not in the document]
Comment (Ruslan05) [reply docs/scos/dev/migration-concepts/migrate-to-paas/step-13-migrate-database.md:70]: You are right. I think we can get rid of it. We have internal legacy script that converts PostgreSQL dump to MariaDB compatible, but we can't share it as it's not something official. But the link itself is not very useful and tested by us. 
So i'd remove this line.

**Pull request merged**